codecov_cli/helpers/ci_adapters/github_actions.py

```diff
@@ -17,14 +20,27 @@ def _get_commit_sha(self):
         if not pr:
             return commit
 
-        # actions/checkout should be run with fetch-depth > 1 or set to 0 for this to work
-        completed_subprocess = subprocess.run(
-            ["git", "rev-parse", "HEAD^@"], capture_output=True
-        )
-
-        parents_hash = completed_subprocess.stdout.decode().strip().splitlines()
-        if len(parents_hash) == 2:
-            return parents_hash[1]
+        merge_commit_regex = r"^[a-z0-9]{40} [a-z0-9]{40}$"
+        merge_commit_message = subprocess.run(
+            ["git", "show", "--no-patch", "--format=%P"],
```

Comment:
You don't actually have to change this adapter's logic. The command being used accounts for this fact and already "fixes" the merge commit if that's the case.

```
$ git rev-parse HEAD^@
1b00e154fb8a8a7ad09da4f9c6c95e70c5fa1565
211b95514359270c24c3e4e2058b3eef83db75e4 <-- this would be returned by the old logic
$ git show --no-patch --format=%P | cat
1b00e154fb8a8a7ad09da4f9c6c95e70c5fa1565 211b95514359270c24c3e4e2058b3eef83db75e4